### ISSUE TITLE
`services.jupyter.enable = true;` fails to evaluate with default settings

### ISSUE THREAD

**Issue by magneticflux-** [NixOS/nixpkgs#253830]
### Describe the bug
Setting `services.jupyter.enable = true;` fails with the message:
```
error:
Failed assertions:
- users.users.jupyter.group is unset. This used to default to
nogroup, but this is unsafe. For example you can create a group
for this user with:
users.users.jupyter.group = "jupyter";
users.groups.jupyter = {};
```

Adding `users.users.jupyter.group = "jupyter";` in the top-level config is a workaround for now.

### Steps To Reproduce
1. Add `services.jupyter.enable = true;` to your `configuration.nix`
2. Run `nixos-rebuild switch --show-trace`
3. Observe error

### Expected behavior
The group should be set to a sane default instead of `{}`.

### Additional context
Relevant code is located here: https://github.com/NixOS/nixpkgs/blob/347b5fb808805277b7797529d518a363fd8483e7/nixos/modules/services/development/jupyter/default.nix#L196-L198

This was not changed after #133166 was merged, causing the error. A similar fix to #138315 should be implemented.

### Notify maintainers

@aborsu maintains `jupyter`, @symphorien has fixed `nogroup` issues in the past.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.14, NixOS, 23.11 (Tapir), 23.11pre523196.347b5fb80880`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by FliegendeWurst** [comment]
Duplicate of #348892
